The failing CI is weird, seems unrelated to me?

Linux
```
/home/build-user/build/Ninja-ReleaseAssert/swift-linux-x86_64/bootstrapping0/lib/swift/linux/libswiftCore.so: error: undefined reference to '__truncsfhf2'
/home/build-user/build/Ninja-ReleaseAssert/swift-linux-x86_64/bootstrapping0/lib/swift/linux/libswiftCore.so: error: undefined reference to '__extendhfsf2'
/home/build-user/build/Ninja-ReleaseAssert/swift-linux-x86_64/bootstrapping0/lib/swift/linux/libswiftCore.so: error: undefined reference to '__extendhfxf2'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

macOS
```
FAILED: lib/swift/macosx/arm64/libswiftRemoteMirror.dylib 
: && /Users/ec2-user/jenkins/workspace/apple-llvm-project-pr-macos/branch-stable/20221013/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/llvm-macosx-x86_64/./bin/clang++ -Wno-unknown-warning-option -Werror=unguarded-availability-new -fno-stack-protector -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -Werror=switch -Wdocumentation -Wimplicit-fallthrough -Wunreachable-code -Woverloaded-virtual -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -O2 -DNDEBUG -arch arm64 -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -dynamiclib -Wl,-headerpad_max_install_names -target arm64-apple-macosx10.9 -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -F/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/../../../Developer/Library/Frameworks -Wl,-dead_strip -Xlinker -compatibility_version -Xlinker 1 -o lib/swift/macosx/arm64/libswiftRemoteMirror.dylib -install_name @rpath/libswiftRemoteMirror.dylib stdlib/public/LLVMSupport/CMakeFiles/swiftLLVMSupport-macosx-arm64.dir/ErrorHandling.cpp.o stdlib/public/LLVMSupport/CMakeFiles/swiftLLVMSupport-macosx-arm64.dir/Hashing.cpp.o stdlib/public/LLVMSupport/CMakeFiles/swiftLLVMSupport-macosx-arm64.dir/MemAlloc.cpp.o stdlib/public/LLVMSupport/CMakeFiles/swiftLLVMSupport-macosx-arm64.dir/SmallPtrSet.cpp.o stdlib/public/LLVMSupport/CMakeFiles/swiftLLVMSupport-macosx-arm64.dir/SmallVector.cpp.o stdlib/public/LLVMSupport/CMakeFiles/swiftLLVMSupport-macosx-arm64.dir/StringRef.cpp.o stdlib/public/SwiftRemoteMirror/CMakeFiles/swiftRemoteMirror-macosx-arm64.dir/SwiftRemoteMirror.cpp.o -L/Users/ec2-user/jenkins/workspace/apple-llvm-project-pr-macos/branch-stable/20221013/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/llvm-macosx-x86_64/./lib   -L/Users/ec2-user/jenkins/workspace/apple-llvm-project-pr-macos/branch-stable/20221013/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/swift-macosx-x86_64/./lib/swift/macosx/arm64   -L/Users/ec2-user/jenkins/workspace/apple-llvm-project-pr-macos/branch-stable/20221013/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/swift-macosx-x86_64/./bin/../lib/swift/macosx/arm64   -L/Users/ec2-user/jenkins/workspace/apple-llvm-project-pr-macos/branch-stable/20221013/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/swift-macosx-x86_64/./bin/../lib/swift/macosx   -L/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/lib/swift lib/swift/macosx/arm64/libswiftReflection.a && :
0  0x100421970  __assert_rtn + 127
1  0x10042b7d2  ld::passes::compact_unwind::UnwindInfoAtom<arm64_32>::addCompressedEncodingFixup(unsigned int, ld::Atom const*) (.cold.1) + 0
2  0x1003d1ceb  ld::passes::compact_unwind::UnwindInfoAtom<arm64>::addCompressedEncodingFixup(unsigned int, ld::Atom const*) + 171
3  0x1003d92c5  ld::passes::compact_unwind::UnwindInfoAtom<arm64>::makeCompressedSecondLevelPage(std::__1::vector<ld::passes::compact_unwind::UnwindEntry, std::__1::allocator<ld::passes::compact_unwind::UnwindEntry> > const&, std::__1::map<unsigned int, unsigned int, std::__1::less<unsigned int>, std::__1::allocator<std::__1::pair<unsigned int const, unsigned int> > >, unsigned int, unsigned int, unsigned char*&) + 1541
4  0x1003d8254  ld::passes::compact_unwind::UnwindInfoAtom<arm64>::UnwindInfoAtom(std::__1::vector<ld::passes::compact_unwind::UnwindEntry, std::__1::allocator<ld::passes::compact_unwind::UnwindEntry> > const&, unsigned long long) + 634
5  0x1003d3626  ld::passes::compact_unwind::doPass(Options const&, ld::Internal&) + 2294
6  0x1002dabb2  main + 616
7  0x7fff20341f3d  start + 1
A linker snapshot was created at:
	/tmp/libswiftRemoteMirror.dylib-2023-00-06-194106.ld-snapshot
ld: Assertion failed: (targetAtom != NULL), function Fixup, file ld.hpp, line 826.
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
```

I had the same ld error on my local machine while building the master, without my changes. I assumed it was because I was using Xcode 14.1, worked on Xcode 13.4.1 for me. It seems the CI is still running a beta version of 13.0 (13A5201i)?

--- a/lldb/source/Core/Mangled.cpp
+++ b/lldb/source/Core/Mangled.cpp
@@ -56,12 +56,8 @@ static inline bool cstring_is_mangled(llvm::StringRef s) {
 // pointers
 static ThreadSafeDenseMap<const char *, ConstString> *
 GetDisplayDemangledNamesCache() {
-  static ThreadSafeDenseMap<const char *, ConstString> *g_cache = nullptr;
-  static std::once_flag g_flag;
-  std::call_once(g_flag, []() -> void {
-    g_cache = new ThreadSafeDenseMap<const char *, ConstString>();
-  });
-  return g_cache;
+  static ThreadSafeDenseMap<const char *, ConstString> g_cache;
+  return &g_cache;
 }
 // END SWIFT
 

--- a/lldb/source/Core/Mangled.cpp
+++ b/lldb/source/Core/Mangled.cpp
@@ -56,9 +56,9 @@ static inline bool cstring_is_mangled(llvm::StringRef s) {
 // pointers
 static ThreadSafeDenseMap<const char *, ConstString> *
 GetDisplayDemangledNamesCache() {
-  ThreadSafeDenseMap<const char *, ConstString> *g_cache;
-  std::once_flag g_flag;
-  std::call_once(g_flag, [&g_cache]() -> void {
+  static ThreadSafeDenseMap<const char *, ConstString> *g_cache = nullptr;
+  static std::once_flag g_flag;
+  std::call_once(g_flag, []() -> void {
     g_cache = new ThreadSafeDenseMap<const char *, ConstString>();
   });
   return g_cache;

--- a/lldb/source/Core/Mangled.cpp
+++ b/lldb/source/Core/Mangled.cpp
@@ -54,10 +54,10 @@ static inline bool cstring_is_mangled(llvm::StringRef s) {
 // invalid value other than ConstString(nullptr) so, just use const
 // char* as the key as LLVM knows how to do proper DenseMapInfo for
 // pointers
-static ThreadSafeDenseMap<const char *, ConstString> *
+static ThreadSafeDenseMap<const char *, ConstString>&
 GetDisplayDemangledNamesCache() {
   static ThreadSafeDenseMap<const char *, ConstString> g_cache;
-  return &g_cache;
+  return g_cache;
 }
 // END SWIFT
 
@@ -372,8 +372,8 @@ ConstString Mangled::GetDisplayDemangledName(
 
       if (mangled) {
         if (SwiftLanguageRuntime::IsSwiftMangledName(m_mangled.GetStringRef())) {
-          auto display_cache = ::GetDisplayDemangledNamesCache();
-          if (display_cache && display_cache->Lookup(mangled, demangled) &&
+          auto& display_cache = ::GetDisplayDemangledNamesCache();
+          if (display_cache.Lookup(mangled, demangled) &&
               demangled)
             break;
 
@@ -383,7 +383,7 @@ ConstString Mangled::GetDisplayDemangledName(
                   sc);
           if (!demangled_std.empty()) {
             demangled.SetCString(demangled_std.c_str());
-            display_cache->Insert(mangled, demangled);
+            display_cache.Insert(mangled, demangled);
             break;
           }
         }

